### PR TITLE
feat: changes for new arch support

### PR DESCRIPTION
**Pull request by dojolew**
Make ios sdk code objc++ for new architecture support

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 ## Installation
 
-Check out our [setup guide 🔧](https://docs.dojo.tech/payments/mobile-integration/react-native/setup) to get started.
+Check out our [setup guide 🔧](https://docs.dojo.tech/mobile-integration/react-native/setup) to get started.
```
Comment (nick-marais-dojo) [inline README.md:17]: Can you fix the link below also?

File: android/gradlew
Comment (nick-marais-dojo) [inline android/gradlew:1]: Why were these added?

**Review by deniss-kaibagarovs — APPROVED** [review]

File: example-new-architecture/ios/.xcode.env.local
Comment (nick-marais-dojo) [inline example-new-architecture/ios/.xcode.env.local:1]: .gitignore?

File: example-new-architecture/ios/File.swift [hunk 2]
```diff
@@ -0,0 +1,9 @@
+//
+//  File.swift
+//  DojoPaySdkExample
+//
+//  Created by Lewis Yearsley on 20/11/2024.
+//
+
+import Foundation
+import dojo_ios_sdk
```
Comment (nick-marais-dojo) [inline example-new-architecture/ios/File.swift:9]: Was this still needed? The line and the file itself?

File: example-new-architecture/src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+const dojoSdkScreenOptions: StackNavigationOptions = {
+  title: 'Dojo Sdk',
+};
+
+const settingsScreenOptions: StackNavigationOptions = {
+  title: 'Settings',
+};
+
+const App: FC = () => {
```
Comment (nick-marais-dojo) [inline example-new-architecture/src/App.tsx:23]: Would be nice to go with a monorepo to share this stuff across example apps - guess when the old one is removed we wont need to worry.

**Review by nick-marais-dojo — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (dojolew) [reply README.md:17]: done

File: example-new-architecture/ios/.xcode.env.local
Comment (dojolew) [reply example-new-architecture/ios/.xcode.env.local:1]: done

File: android/gradlew
Comment (dojolew) [reply android/gradlew:1]: gradlew is a commited file, the existing example app does and also our Dojo app

File: example-new-architecture/ios/File.swift (hunk 2 above)
Comment (dojolew) [reply example-new-architecture/ios/File.swift:9]: Removed

File: example-new-architecture/src/App.tsx (hunk 3 above)
Comment (dojolew) [reply example-new-architecture/src/App.tsx:23]: It would, something to think about down the line if old arch proves to stick around for a while - we should look into some branding in the example as well at some point

File: android/gradlew
Comment (nick-marais-dojo) [reply android/gradlew:1]: These are needed for an "app", but this is the lib.

Comment (dojolew) [reply android/gradlew:1]: fixed
